easycv/pipeline.py

```diff
@@ -26,78 +31,206 @@ class Pipeline:
 
     def __init__(self, source, name=None):
         if isinstance(source, list):
-            self.forwards = Pipeline._calculate_forwards(source)
+            self.forwards = {}
 
             self.arguments = source[0].arguments if source else {}
             self.outputs = source[-1].outputs if source else {}
 
             self._name = name if name else "pipeline"
             self._transforms = deepcopy(source)
+            self.required = {}
+            self.optional = {}
+            self._arguments = {}
+            self.initialize()
 
         elif isinstance(source, str) and os.path.isfile(source):
-            try:
-                with open(source, "rb") as f:
-                    saved = pickle.load(f)
-                    if isinstance(saved, Pipeline):
-                        self._name = name if name else saved.name
-                        self._transforms = saved.transforms()
-                    else:
-                        raise InvalidPipelineInputSource()
-            except pickle.UnpicklingError:
-                raise InvalidPipelineInputSource() from None
+            with open(source, "r") as file:
+                source = yaml.load(file, Loader=yaml.FullLoader)
+            self.from_dict(source)
+        elif isinstance(source, dict):
+            self.from_dict(source)
         else:
             raise InvalidPipelineInputSource()
 
-    @staticmethod
-    def _calculate_forwards(source):
-        outputs = {}
-        forwards = {}
-
-        for i in range(len(source)):
-            forwards[i] = {}
+    def from_dict(self, source):
+        transforms = []
+        pipe = source["Pipeline"]
+        self.forwards = pipe["Forwards"]
+        self._name = pipe["name"]
+        for operation in pipe["Operations"]:
+            if "Transform" in operation:
+                transform = get_transform(operation["Transform"]["name"])
+                arguments = {
+                    arg: operation["Transform"]["args"][arg]
+                    for arg in operation["Transform"]["args"]
+                    if operation["Transform"]["args"][arg]
+                }
+                transforms.append(
+                    transform(
+                        method=operation["Transform"]["method"],
+                        r_in=operation["Transform"]["renamed_in"],
+                        r_out=operation["Transform"]["renamed_out"],
+                        **arguments
+                    )
+                )
+            elif "Pipeline" in operation:
+                transforms.append(Pipeline(operation))
+            else:
+                raise MissingArgumentsError()
```

Comment:
Codacy found an issue: [No value for argument 'arg' in constructor call](https://app.codacy.com/gh/easycv/easycv/pullRequest?prid=6255509)